apps/bitwarden-mailconfig.sh

```diff
@@ -0,0 +1,218 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+# Must be root
+root_check
+
+# Check if Bitwarden is already installed
+print_text_in_color "$ICyan" "Checking if Bitwarden is already installed..."
+if is_docker_running
+then
+    if docker ps -a --format '{{.Names}}' | grep -Eq "bitwarden";
+    then
+        if [ ! -d "$BITWARDEN_HOME"/bwdata ]
+        then
+            msg_box "It seems like 'Bitwarden' isn't installed in the new place.\n\nYou cannot run this script."
+            exit 1
+        fi
+    else
+        msg_box "It seems like 'Bitwarden' isn't installed.\n\nYou cannot run this script."
+        exit 1
+    fi
+else
+    msg_box "It seems like 'Bitwarden' isn't installed.\n\nYou cannot run this script."
+    exit 1
+fi
+
+msg_box "This script lets you configure your mailserver settings for Bitwarden."
+if [[ "no" == $(ask_yes_or_no "Do you want to continue?") ]]
+then
+    exit
+fi
+
+# Insert globalSettings__mail__smtp__trustServer to global.override
+if ! grep -q "^globalSettings__mail__smtp__trustServer=" "$BITWARDEN_HOME"/bwdata/env/global.override.env
+then
+    echo "globalSettings__mail__smtp__trustServer=false" >> "$BITWARDEN_HOME"/bwdata/env/global.override.env
+fi
+
+# Enter Mailserver
+while true
+do
+    MAIL_SERVER=$(whiptail --inputbox "Please enter the Mailserver, that you want to use.\nE.g. smtp.mail.de\nIf you don't want to change the Mailserver, that is already configured inside the global.override.env-file, just leave the box empty." "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Are your sure about your answer?") ]]
+    then
+        msg_box "It seems like your weren't satisfied by the Mailserver you entered. Please try again."
+    else
+        break
+    fi
+done
+
+# Enter if you want to use ssl
+while true
+do
+    USE_SSL=$(whiptail --inputbox "Do you want to use SSL for your Mailserver? If yes: please type in 'yes'.\nPlease note: if your Mailserver only supports Starttls, you cannot use SSL - so please answer in this case here 'no'.\nIf you don't want to change the SSL-setting, that is already configured inside the global.override.env-file, just leave the box empty." "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
```

Comment:
SSL here is port 465, that should be clear.

Reply:
Hm, I am not sure about this. Isn't there a possibility where you can enable this on another Port?

Reply:
Isn't 465 hardcoded?

Reply:
No, it isn't. 
I just use the Default_port variable to suggest a Port based on if he wants to use ssl or not...

Reply:
Aaah, to tired now. :smile: Well, I think 587 (TLS) should be the default.

Reply:
You can choose the Port already independently from the used encryption protocol.